README.md

```diff
@@ -9,6 +9,65 @@ go get github.com/sourcegraph/sourcegraph-accounts-sdk-go
 > [!note]
 > Please submit all issues to the [`sourcegraph/sourcegraph-accounts` repository](https://github.com/sourcegraph/sourcegraph-accounts/issues) 
 
+## Authentication
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	samsauth "github.com/sourcegraph/sourcegraph-accounts-sdk-go/auth"
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/scopes"
+)
+
+func main() {
+	stateSetter := func(w http.ResponseWriter, r *http.Request, state string) error {
+		// TODO: Save to session data. For best security measures, store in a backend
+		// component (e.g. Redis) or encrypted cookie. Unencrypted cookies are strongly
+		// discouraged as they can be tampered with.
+		return nil
+	}
+	stateGetter := func(r *http.Request) (string, error) {
+		// TODO: Retrieve from session data.
+		return "", nil
+	}
+	stateDeleter := func(w http.ResponseWriter, r *http.Request) {
+		// TODO: Delete from session data.
+	}
+
+	samsauthHandler, err := samsauth.NewHandler(
+		samsauth.Config{
+			Issuer:         "https://accounts.sourcegraph.com",
+			ClientID:       os.Getenv("SAMS_CLIENT_ID"),
+			ClientSecret:   os.Getenv("SAMS_CLIENT_SECRET"),
+			RequestScopes:  []scopes.Scope{scopes.OpenID, scopes.Email, scopes.Profile},
```

Comment:
> What is the significance of these scopes here? I am guessing that these are the scopes/permissions that the user would be authorizing when direct to SAMS, right? e.g. "the application 'cody backend' would like to have access to your user profile, and subscription:write"?

Yes, however since all SAMS clients are internal/official clients, we do not show the consent screen to users at all (which we may in the far future).



> This is probably worth calling out in a comment. Since it's an important configuration knob. (Because anything the person using this SAMS login flow will need to know what scopes the user would have access to.) something something "when using the refresh token and generating an access token" the OAuth RFC says you can only request a subset of the scopes on the refresh token or something?

Sure we can add a quick comment mentioning this, although we are really only using/having default OIDC scopes for user authentication (to get user profile info), for the sake of OIDC-compliant. Due to the fact we provide first-class Clients API, as long as the client has granted scopes registered in the SAMS database, clients can always ask for any granted scopes for any access tokens because that's not doing things on behalf of the user (unlike GitHub APIs) but on behalf of the client itself.